sdk-base/src/main/java/com/mapbox/maps/plugin/viewport/experimental/state/ViewportDataSourceObserver.kt

```diff
@@ -0,0 +1,16 @@
+package com.mapbox.maps.plugin.viewport.experimental.state
+
+import com.mapbox.maps.CameraOptions
+
+/**
+ * Observer that gets notified whenever new data is available.
+ */
+fun interface ViewportDataSourceObserver {
+  /**
+   * Notifies that new data is available.
+   *
+   * @param cameraOptions The latest [CameraOptions] from the [ViewportState].
+   * @return true if new data is needed. returning false will unsubscribe from further data updates.
+   */
+  fun onNewData(cameraOptions: CameraOptions): Boolean
```

Comment:
Maybe rename to 
`onNewViewport` / `onNewCameraOptions` ?
Also, class could also be named `ViewportObserver`/`CameraOptionsObserver`

Reply:
Renamed the class to `ViewportStateDataObserver`, as ViewportObserver and CameraOptionsObserver seems too generic, this observer is used to observe the ViewportState's data updates.